Instant-Messenger/src/server/server.cpp

```diff
@@ -312,4 +359,199 @@ namespace server {
         return socketId;
     }
 
+
+    void Server::connectToRmServers() {
+        for(int connectMachineNumber = rmNumber; connectMachineNumber < MAX_RM; connectMachineNumber++) {
+            int *connectSocket = (int *) calloc(1, sizeof(int *));
+            char ip_address[10] = "127.0.0.1";
+            struct sockaddr_in rm_connect_socket_addr{};
+
+            rm_connect_socket_addr.sin_family = AF_INET;
+            rm_connect_socket_addr.sin_port = htons(RM_BASE_PORT_NUMBER + connectMachineNumber + 1); // port is base port + next machine RM number (rmNumber+1)
+            bzero(&(rm_connect_socket_addr.sin_zero), 8);
+
+            if (inet_pton(AF_INET, ip_address, &rm_connect_socket_addr.sin_addr) <= 0) {
+                cout << "\n RM Invalid address/ Address not supported \n" << endl;
+                continue;
+            }
+
+            if ((*connectSocket = socket(AF_INET, SOCK_STREAM, 0)) < 0) {
+                cout << "\n RM Socket creation error \n" << endl;
+                continue;
+            }
+
+            if (connect(*connectSocket, (struct sockaddr *) &rm_connect_socket_addr, sizeof(rm_connect_socket_addr)) < 0) {
+                cout << "ERROR connecting on RM sockets\n" << endl;
+                continue;
+            }
+
+            rm_connect_sockets_fd.insert({*connectSocket, rm_connect_socket_addr});
+            cout << "Connected to socket " << *connectSocket << " and port " << ntohs(rm_connect_socket_addr.sin_port) << endl;
+
+            std::pair<int *, Server *> *args = (std::pair<int *, Server *> *) calloc(1, sizeof(std::pair<int *, Server *>));
+
+            // Send pointer of the previously allocated address and be able to access it's value in new thread's execution
+            args->first = connectSocket;
+            args->second = this;
+
+            pthread_t connectedRMThread;
+            pthread_create(&connectedRMThread, NULL, handleRMCommunication, (void *) args);
+        }
+    }
+
+/**
+* This method creates listener sockets based on RM number
+*/
+    void Server::createRMListenerSocket() {
+        int opt = 1;
+        int rmListeningPort = RM_BASE_PORT_NUMBER + rmNumber;
+        rm_listening_serv_addr.sin_port = htons(rmListeningPort);
+
+        // Create socket file descriptor
+        if ((rm_listening_socket_fd = socket(AF_INET, SOCK_STREAM, 0)) <= 0) {
+            cout << "ERROR opening socket\n" << endl;
+            exit(1);
+        }
+
+        // Forcefully attaching socket to the port
+        if (setsockopt(rm_listening_socket_fd, SOL_SOCKET, SO_REUSEADDR, &opt, sizeof(opt)))
```

Comment:
Use it as defined by Alberto and previous etapa

Reply:
Previous etapa é foda aeuhueahuea

Reply:
Same issue as above discussion. No support for MacOS systems.